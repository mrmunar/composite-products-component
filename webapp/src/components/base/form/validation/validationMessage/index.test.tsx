import React from 'react';
import { render, cleanup, fireEvent, screen } from '@testing-library/react';
import ValidationMessage from './';

afterEach(cleanup);

test('should take a snapshot', () => {
    const { baseElement } = render(<ValidationMessage />);
    expect(baseElement).toMatchSnapshot();
});

test('return null if valid', () => {
    const { queryAllByTestId } = render(<ValidationMessage valid={true} />);
    expect(screen.queryByTestId('validation-msg')).toBeNull()
});

test('return null if valid', () => {
    const { queryAllByTestId } = render(<ValidationMessage valid={false} />);
    expect(screen.queryByTestId('validation-msg')).not.toBeNull()
});

test('can have a message value', () => {
    const { getByTestId } = render(<ValidationMessage valid={false} message="Test Message" />);
    expect(getByTestId('validation-msg')).toHaveTextContent('Test Message');
});


