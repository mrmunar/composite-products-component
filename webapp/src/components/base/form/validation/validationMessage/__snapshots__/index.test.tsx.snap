// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<body>
  <div>
    <div
      class="p-1 m-2 alert alert-danger"
      data-testid="validation-msg"
    />
  </div>
</body>
`;
